### PR TITLE
WIP on Issue 49

### PR DESCRIPTION
**Pull request by Sandeep-1-Kumar**
Added method signatures just to satisfy TDD in page.mjs
-asCanvas returns Annotation added.
-create added
-fetch added
-remove added
-save added

Corrected typo in exists.test.mjs

### CODE REVIEW

File: classes/Page/Page.mjs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    }
+
+    set id(id) {
+        this.page.id = id
+    }
+
+
+    asCanvas() {
+        return {
+            type: 'Annotation',
```
Comment (cubap) [inline classes/Page/Page.mjs:20]: Look at the "Canvas" object for basic shape of a Canvas at https://iiif.io/api/presentation/3.0/

Comment (Sandeep-1-Kumar) [reply classes/Page/Page.mjs]: Updated on this , I had gone through the document and added the required properties.

File: classes/Page/Page.mjs [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+        return {
+            type: 'Canvas',
+            '@context': 'http://iiif.io/api/presentation/3/context.json',
+            id:this.page.id ?? 0,
+            height: this.page.height ?? 0,
+            width: this.page.width ?? 0,
+            label: this.page.label ?? '',
+            summary: this.page.summary ?? '',
+            thumbnail: this.page.thumbnail ?? '',
+            navDate:this.page.navDate ?? '',
```
Comment (cubap) [inline classes/Page/Page.mjs:28]: some of these may be more appropriate to leave off if not present, especially if not mandatory by the IIIF spec

File: classes/Page/Page.mjs [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+        return Promise.resolve()
+    }
+    
+    // Method to fetch a page
+    fetch() {
+        return Promise.resolve(new Page())
+    }
+
+    // Method to add membership reference to manifest
+    addMembershipToManifest() {
```
Comment (cubap) [inline classes/Page/Page.mjs:59]: this was unclear in the docs, I know. This is to get the Manifest container object this Page is in/partOf

File: classes/Page/Page.mjs [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+    }
+
+    // Method to get the next page
+    getNextPage() {
+        return Promise.resolve(new Page())
+    }
+
+
+    // Method to handle text blob
+    handleTextBlob(blob) {
```
Comment (cubap) [inline classes/Page/Page.mjs:75]: getTextContent() as a continuous concatenated string from the Line.body or as an Array of Lines.body. If you are imagining a different behavior, describe it in the comment.

File: classes/Page/Page.mjs [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+        return Promise.resolve('Text blob handled')
+    }
+
+    // Method to handle image annotation
+    handleImageAnnotation(url) {
```
Comment (cubap) [inline classes/Page/Page.mjs:80]: return the image on this page that is being annotated as full or an array of URL fragments per Line (like text above)

File: classes/Page/Page.mjs [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+        return Promise.resolve('image annotation handled')
+    }
+
+    // Method to get parent page
+    getParentPage() {
```
Comment (cubap) [inline classes/Page/Page.mjs:85]: Page has no parent

File: classes/Page/Page.mjs [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+
+    // Method to get project
+    getProject() {
+        return Promise.resolve('Project retrieved')
+    }
+
+    // Method to get parent/sibling/children/project information
+    getPageInfo(type) {
+        switch(type) {
+            case 'parent':
```
Comment (cubap) [inline classes/Page/Page.mjs:107]: no parent/children

File: classes/Page/Page.mjs [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+        return Promise.resolve('Documents embedded')
+    }
+
+    // Method to retrieve metadata only
+    getMetadata() {
+        return Promise.resolve('Metadata of the page');
+    }
+
+    // Method to retrieve HTML document
+    getHTMLDocument() {
```
Comment (cubap) [inline classes/Page/Page.mjs:131]: HTML version of `asJSON()` so "asHTML()" might be better

**Review by cubap — CHANGES_REQUESTED** [review]
Close. I'll put some general Slack comments too

File: classes/Page/Page.mjs (hunk 2 above)
Comment (Sandeep-1-Kumar) [reply classes/Page/Page.mjs:28]: Can you please be more specific on this please!!

File: classes/Page/Page.mjs (hunk 8 above)
Comment (Sandeep-1-Kumar) [reply classes/Page/Page.mjs:131]: Resolved.

File: classes/Page/Page.mjs [hunk 9]
```diff
@@ -0,0 +1,135 @@
...
+                return Promise.resolve('Children pages information retrieved')
+            case 'project':
+                return Promise.resolve('Project information retrieved')
+            default:
+                throw new Error('Invalid type specified. Valid types are: parent, siblings, children, project');
+        }
+    }
+
+    // Method to embed referenced documents
+    embedDocuments() {
```
[the remark this answers is not in the document]
Comment (Sandeep-1-Kumar) [reply classes/Page/Page.mjs:121]: removed embedDocuments() method.

File: classes/Page/Page.mjs (hunk 7 above)
Comment (Sandeep-1-Kumar) [reply classes/Page/Page.mjs:107]: removed, resolved.

File: classes/Page/Page.mjs [hunk 10]
```diff
@@ -0,0 +1,135 @@
...
+        return Promise.resolve('Parent page retrieved')
+    }
+
+    // Method to get sibling pages
+    getSiblingPages() {
+        return Promise.resolve('sibling pages retrieved')
+    }
+
+    // Method to get children pages
+    getChildrenPages() {
```
[the remark this answers is not in the document]
Comment (Sandeep-1-Kumar) [reply classes/Page/Page.mjs:95]: removed, resolved.

File: classes/Page/Page.mjs (hunk 6 above)
Comment (Sandeep-1-Kumar) [reply classes/Page/Page.mjs:85]: removed, resolved.

File: classes/Page/Page.mjs (hunk 5 above)
Comment (Sandeep-1-Kumar) [reply classes/Page/Page.mjs:80]: corrected.

File: classes/Page/Page.mjs (hunk 4 above)
Comment (Sandeep-1-Kumar) [reply classes/Page/Page.mjs:75]: Corrected resolved.

File: classes/Page/Page.mjs (hunk 3 above)
Comment (Sandeep-1-Kumar) [reply classes/Page/Page.mjs:59]: updated , resolved.

File: classes/Page/Page.mjs (hunk 2 above)
Comment (Sandeep-1-Kumar) [reply classes/Page/Page.mjs]: I had just commented out the fields , Once i have more clarity on the properties will neatly deal with it.

**Review by cubap — APPROVED** [review]

**Review by BhanuKandula5030 — APPROVED** [review]
Looks good !

**Review by leandrumartin — APPROVED** [review]

**Pull request merged**